remote-api/src/main/java/org/alfresco/rest/api/impl/rules/RulesImpl.java

```diff
@@ -130,6 +131,20 @@ private org.alfresco.service.cmr.rule.Rule mapToServiceModelAndValidateActions(R
         return actionPermissionValidator.validateRulePermissions(serviceModelRule);
     }
 
+    private Rule loadRuleAndConvertActionParams(org.alfresco.service.cmr.rule.Rule ruleModel, List<String> includes)
+    {
+
+        final Rule rule = ruleLoader.loadRule(ruleModel, includes);
+        rule.getActions()
```

Comment:
Minor, this conversion of params could be done within `Action.from()` so it would be made in one place, but I'm also ok with this approach.

Reply:
I am not happy with approach either but IMO, all those mappings (`Rule`, `Action`, `Condition` etc.) should be done in mapper classes responsible only for mapping to/from layers. 
That way we could inject mappers and additional dependencies inside mappers via DI framework mechanisms instead of passing them as method parameters (as for instance in `Rule::toServiceModel`).
If I were to map the parameters via `Action.from()` I'd have to pass dependencies via method parameters through 2 or 3 methods (`Rule` -> `CompositeAction` -> `Action`) which I think is a worse solution.

Reply:
I agree with that, intermediate mapping layer would be much better approach.